src/Microsoft.Identity.Client/Mats/Internal/ActionAggregator.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Identity.Client.Mats.Internal
+{
+    internal static class ActionAggregator

```

Comment:
All these action classes are pretty complex. Would be great to add class comments that focus on some small examples.